Microsoft.Toolkit.Win32/Microsoft.Toolkit.Win32.UI.Controls/Interop/WinRT/WebViewControlNavigationStartingEventArgs.cs

```diff
@@ -10,38 +10,59 @@ namespace Microsoft.Toolkit.Win32.UI.Controls.Interop.WinRT
     /// <summary>
     /// Provides data for the <see cref="IWebView.NavigationStarting"/> and <see cref="IWebView.FrameNavigationStarting"/> events. This class cannot be inherited.
     /// </summary>
-    /// <remarks>Copy from <see cref="Windows.Web.UI.WebViewControlNavigationStartingEventArgs"/> to avoid requirement to link Windows.winmd</remarks>
+    /// <remarks>Copy from <see cref="global::Windows.Web.UI.WebViewControlNavigationStartingEventArgs"/> to avoid requirement to link Windows.winmd</remarks>
     /// <seealso cref="System.EventArgs" />
-    /// <seealso cref="Windows.Web.UI.WebViewControlNavigationStartingEventArgs"/>
+    /// <seealso cref="global::Windows.Web.UI.WebViewControlNavigationStartingEventArgs"/>
     public sealed class WebViewControlNavigationStartingEventArgs : EventArgs
     {
         [SecurityCritical]
-        private readonly Windows.Web.UI.WebViewControlNavigationStartingEventArgs _args;
+        private readonly global::Windows.Web.UI.WebViewControlNavigationStartingEventArgs _args;
 
         [SecurityCritical]
-        internal WebViewControlNavigationStartingEventArgs(Windows.Web.UI.WebViewControlNavigationStartingEventArgs args)
+        private readonly System.Windows.Navigation.NavigatingCancelEventArgs _compatibleArgs;
+
+        [SecurityCritical]
+        internal WebViewControlNavigationStartingEventArgs(global::Windows.Web.UI.WebViewControlNavigationStartingEventArgs args)
         {
             _args = args;
             Uri = args.Uri;
         }
 
         [SecurityCritical]
-        internal WebViewControlNavigationStartingEventArgs(Windows.Web.UI.WebViewControlNavigationStartingEventArgs args, Uri uri)
+        internal WebViewControlNavigationStartingEventArgs(global::Windows.Web.UI.WebViewControlNavigationStartingEventArgs args, Uri uri)
             : this(args)
         {
             Uri = uri;
         }
 
+        [SecurityCritical]
+        internal WebViewControlNavigationStartingEventArgs(System.Windows.Navigation.NavigatingCancelEventArgs e)
+        {
+            _args = null;
+            _compatibleArgs = e;
```

Comment:
This pattern exists in a couple of places (so far). I'm confused why this is necessary. Causes complexity in accessors like Cancel below

Reply:
I agree.  In quite a few cases, I replicated existing patterns without question in order to focus on to more important things... figuring if it's important enough to change, it'll surface during a review...

Reply:
Ok... in looking at fixing it, I remember that the pattern serves a purpose...  there are cases where we need to raise the event, but don't have a UWP-created instance to wrap.  Since UWP event args classes are internal / sealed with private c'tor's, we don't have anything we can substitute with, so we need the extra complexity.

Reply:
See e4241694171b347b4877b513f1faf374333dfeae as an example of how to implement single responsibility